docker-compose.yml

```diff
@@ -1,69 +1,128 @@
-# NOTE: This docker-compose.yml is meant to be just an example of how
-# you could accomplish this on your own. It is not intended to work in
-# all use-cases and must be adapted to fit your needs. This is merely
-# a guideline.
-
-# See docs.getsentry.com/on-premise/server/ for full
-# instructions
-
 version: '3.4'
-
-x-defaults: &defaults
+x-defaults:
   restart: unless-stopped
-  build:
+  build: &ref_0
     context: .
     args:
       SENTRY_IMAGE: ${SENTRY_IMAGE}
-  depends_on:
+  depends_on: &ref_1
     - redis
     - postgres
     - memcached
     - smtp
+    - snuba
+    - symbolicator
+    - kafka
   env_file: .env
-  environment:
+  environment: &ref_2
     SENTRY_MEMCACHED_HOST: memcached
     SENTRY_REDIS_HOST: redis
     SENTRY_POSTGRES_HOST: postgres
     SENTRY_EMAIL_HOST: smtp
-  volumes:
-    - sentry-data:/var/lib/sentry/files
-
-
+    SNUBA: 'http://snuba:1218'
+    SENTRY_KAFKA_HOST: 'kafka:9092'
+  volumes: &ref_3
+    - 'sentry-data:/var/lib/sentry/files'
 services:
   smtp:
     restart: unless-stopped
     image: tianon/exim4
-
   memcached:
     restart: unless-stopped
-    image: memcached:1.5-alpine
-
+    image: 'memcached:1.5-alpine'
   redis:
     restart: unless-stopped
-    image: redis:3.2-alpine
-
+    image: 'redis:5.0-alpine'
+    command: redis-server --appendonly yes
+    volumes:
+      - 'sentry-redis:/data'
   postgres:
     restart: unless-stopped
-    image: postgres:9.5
+    image: 'postgres:9.6-alpine'
+    volumes:
+      - 'sentry-postgres:/var/lib/postgresql/data'
+  zookeeper:
+    image: 'confluentinc/cp-zookeeper:5.1.2'
+    environment:
+      ZOOKEEPER_CLIENT_PORT: '2181'
+    volumes:
+      - 'sentry-zookeeper:/var/lib/zookeeper'
+  kafka:
+    depends_on:
+      - zookeeper
+    image: 'confluentinc/cp-kafka:5.1.2'
+    environment:
+      KAFKA_ZOOKEEPER_CONNECT: 'zookeeper:2181'
+      KAFKA_LISTENERS: 'INTERNAL://kafka:9093,EXTERNAL://kafka:9092'
+      KAFKA_ADVERTISED_LISTENERS: 'INTERNAL://kafka:9093,EXTERNAL://kafka:9092'
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: 'INTERNAL:PLAINTEXT,EXTERNAL:PLAINTEXT'
+      KAFKA_INTER_BROKER_LISTENER_NAME: INTERNAL
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: '1'
+    volumes:
+      - 'sentry-kafka:/var/lib/kafka'
+  clickhouse:
+    image: 'yandex/clickhouse-server:19.3'
+    ulimits:
+      nofile:
+        soft: 262144
+        hard: 262144
     volumes:
-      - sentry-postgres:/var/lib/postgresql/data
-
+      - 'sentry-clickhouse:/var/lib/clickhouse'
+  snuba:
+    depends_on:
+      - redis
+      - clickhouse
+      - kafka
+    restart: unless-stopped
+    image: 'getsentry/snuba:latest'
+    environment:
+      PYTHONUNBUFFERED: '1'
+      SNUBA_SETTINGS: docker
+      DEBUG: '1'
+      CLICKHOUSE_TABLE: dev
+      CLICKHOUSE_SERVER: 'clickhouse:9000'
+      DEFAULT_BROKERS: 'kafka:9093'
+      REDIS_HOST: redis
+      REDIS_PORT: '6379'
+      REDIS_DB: '1'
+  symbolicator:
```

Comment:
Is this really all that's needed for symbolicator?

Reply:
Calling @jan-auer as I have no idea. I copied most of this from `sentry devservices up`.

Reply:
Let's dig this out of what we do in production first, and we'll go from there. I need to also run this in single-tenant this week, so maybe wait until then and we can share.

Reply:
For prod, there needs to be more configuration. You can have a look at the prod config [here](
https://github.com/getsentry/ops/blob/9b26058709f2f186c5f11d22faa90f261a10d9cf/cookbooks/symbolicator/attributes/default.rb) and the recipe [here](https://github.com/getsentry/ops/blob/07cb861578d46580678db25728a0eb9caad54cad/cookbooks/symbolicator/recipes/default.rb).

Here's the important bits:
- Create a config file somewhere, bind it into the container and pass it to `run --config <path>`
- Bind a storage volume and configure the `cache_dir` to point into that location. Symbolicator stores files debug files in there. Without this, it's basically unusable.
- Configure the `bind` address, the suggested one would be `0.0.0.0:3021`.
- All other configuration can be omitted, unless you want to specifically create debug logs on stdout or log to a statsd server.

Now to [configuring Sentry](https://github.com/getsentry/symbolicator#usage-with-sentry):
 - The IP address of the symbolicator container must be white-listed in Sentry using `INTERNAL_SYSTEM_IPS`.
 - Set [these](https://github.com/getsentry/getsentry/blob/a049b8b096ecfa8a6f0329c791caf3f8c56a4a62/getsentry/conf/settings/prod.py#L375-L379) options accordingly (see below):
- `system.internal-url-prefix` is where Symbolicator can reach Sentry's API, so it will probably be something like `web:9000`
- `symbolicator.enabled` nees to be `True` and `symbolicator.options` needs to point to symbolicator and it's port, so probably `symbolicator:3021`.